totoify_grafeas/totoifylib.py

```diff
@@ -0,0 +1,277 @@
+    resource: A dict of resource uri.
+    intoto: A dictionary with key "signatures", a list of signatures consisting
+        of key id and signature hash, and key "signed", the GrafeasLink class.
+  """
+  note_name = None
+  kind = "INTOTO"
+  resource = {
+    "uri": None
+  }
+  intoto = {
+    "signatures":  [],
+    "signed": None
+  }
+
+  def __init__(self, in_toto_link=None, note_name=None, resource_uri=None):
+    """Initalizes intoto "signed" fields with a GrafeasLink."""
+    if in_toto_link:
+      self.intoto["signed"] = GrafeasLink(in_toto_link.signed.materials,
+                                          in_toto_link.signed.products,
+                                          in_toto_link.signed.command,
+                                          in_toto_link.signed.byproducts,
+                                          in_toto_link.signed.environment)
+      for signature in in_toto_link.signatures:
+        self.intoto["signatures"].append({"keyid": signature["keyid"], "signature": signature["sig"]})
+
+    self.note_name = note_name
+
+    self.resource["uri"] = resource_uri
+
+  def to_json(self, file_path=None):
+    """Returns the JSON string representation."""
+    if file_path:
+      with open(file_path, "w") as fp:
+        json.dump({
+            "resource": self.resource,
+            "noteName": self.note_name,
+            "kind": self.kind,
+            "intoto": {
+              "signatures": self.intoto["signatures"],
+              "signed": self.intoto["signed"].to_dict()
+            }
+          }, fp, indent=4)
+    else:
+      return json.dumps({
+          "resource": self.resource,
+          "noteName": self.note_name,
+          "kind": self.kind,
+          "intoto": {
+            "signatures": self.intoto["signatures"],
+            "signed": self.intoto["signed"].to_dict()
+          }
+        }
+      )
+
+  @staticmethod
+  def load(path):
+    """Given a path, loads the occurrence.json into GrafeasInTotoOccurrence class."""
+    with open(path, "r") as occ_f:
+      occ_json = json.load(occ_f)
+
+    grafeas_occurrence = GrafeasInTotoOccurrence()
+    grafeas_occurrence.intoto['signed'] = GrafeasLink(occ_json['intoto']['signed']['materials'],
+                                                      occ_json['intoto']['signed']['products'],
+                                                      occ_json['intoto']['signed']['command'],
+                                                      occ_json['intoto']['signed']['byproducts'],
+                                                      occ_json['intoto']['signed']['environment'])
+    grafeas_occurrence.intoto['signatures'] = occ_json['intoto']['signatures']
+    grafeas_occurrence.resource['uri'] = occ_json['resource']['uri']
+    grafeas_occurrence.noteName = occ_json['noteName']
+    return grafeas_occurrence
+
+
+
+def create_grafeas_occurrence_from_in_toto_link(in_toto_link, step_name, resource_uri):
```

Comment:
I'm assuming this is a static method. I wonder if we'd like something a little more succinct for this method name. I'm thinking of:

```
grafeas_link = GrafeasInTotoOcurrence.from_link(link)
```

Should make things readable and short...